runtime/vm/ClassInitialization.cpp

```diff
@@ -796,6 +796,14 @@ classInitStateMachine(J9VMThread *currentThread, J9Class *clazz, J9ClassInitStat
 					clazz = VM_VMHelpers::currentClass(clazz);
 					goto done;
 				}
+#if defined(J9VM_OPT_VALHALLA_STRICT_FIELDS)
+				/* All strict static fields must be set by the end of <clinit> */
+				if (clazz->strictStaticFieldCounter > 0) {
+					J9UTF8 *className = J9ROMCLASS_CLASSNAME(clazz->romClass);
+					setCurrentExceptionNLSWithArgs(currentThread, J9NLS_VM_CLASS_LOADING_ERROR_STRICT_STATIC_FIELDS_UNSET, J9VMCONSTANTPOOL_JAVALANGILLEGALSTATEEXCEPTION, J9UTF8_DATA(className));
```

Comment:
This line is overly long. Also, class names are not necessarily NUL-terminated, so the template must accept the length:
```c
					setCurrentExceptionNLSWithArgs(
							currentThread,
							J9NLS_VM_CLASS_LOADING_ERROR_STRICT_STATIC_FIELDS_UNSET,
							J9VMCONSTANTPOOL_JAVALANGILLEGALSTATEEXCEPTION,
							J9UTF8_LENGTH(className),
							J9UTF8_DATA(className));
```